Does not compile with make on macos

I found that when I run make on my apple silicon mac ($OSTYPE==darwin22), I get an error:
 make: *** [shlib] Error 1
I found that this line is the one it fails on:  
@test -f lz4/lib/liblz4.$(SHAREDEXT) && cp lz4/lib/liblz4.$(SHAREDEXT) slib/.
I found that $SHAREDEXT should be dylib, but it is being set to so.  The issue is with these lines

SHAREDEXT=so
ifneq (,$(findstring darwin,$(OSTYPE)))
  SHAREDEXT=dylib
endif

I found that hardcoding the SHAREDEXT to dylib fixes the issue for my mac.  
Somehow it's not setting the SHAREDEXT to dylib

--- a/Makefile
+++ b/Makefile
@@ -11,15 +11,30 @@
 #   === Jefferson National Lab (2017)                      *
 #***********************************************************
 #
-SHAREDEXT=so
 
-ifneq (,$(findstring darwin,$(OSTYPE)))
-  SHAREDEXT=dylib
+### Detect OS
+ifeq ($(OS),Windows_NT)
+  $(info OS detection: Windows)
+  $(error Windows is not supported for make, try cmake)
 endif
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Linux)
+  $(info OS detection: Linux)
+  SHAREDEXT = so
+endif
+ifeq ($(UNAME_S),Darwin)
+  $(info OS detection: Darwin)
+  SHAREDEXT = dylib
+endif
+ifeq ($(SHAREDEXT),)
+  $(warning OS detection failed; assuming Linux)
+  SHAREDEXT = so
+endif
+
 
 all: lib shlib
 
-lib: 
+lib:
 	@test -d lib || mkdir -p lib
 	@cd lz4 ; make lz4
 	@cp lz4/lib/liblz4.a lib/.

